libs/vm_monitor/src/runner.rs

```diff
@@ -62,16 +66,40 @@ pub struct Config {
     /// upscale resource amounts (because we might not *actually* have been upscaled yet). This field
     /// should be removed once we have a better solution there.
     sys_buffer_bytes: u64,
+
+    /// Minimum fraction of total system memory that the threshold for cgroup non-reclaimable
+    /// memory usage must be below
+    cgroup_min_overhead_fraction: f64,
+
+    cgroup_downscale_threshold_buffer_bytes: u64,
 }
 
 impl Default for Config {
     fn default() -> Self {
         Self {
             sys_buffer_bytes: 100 * MiB,
+            cgroup_min_overhead_fraction: 0.15,
+            cgroup_downscale_threshold_buffer_bytes: 100 * MiB,
         }
     }
 }
 
+impl Config {
+    fn cgroup_threshold(&self, total_mem: u64, file_cache_disk_size: u64) -> u64 {
+        // If the file cache is in memory, then it will count towards shmem usage of the cgroup,
+        // and thus be non-reclaimable, so we should allow for additional memory usage. Otherwise
+        // (i.e. if the file cache is on disk), the cgroup's non-reclaimable memory usage will not
+        // include the file cache, and so non-reclaimable memory usage from the cgroup *plus*
+        // memory reserved for the file cache should try to be less than total memory.
+        let memory_remaining_for_cgroup = total_mem.saturating_sub(file_cache_disk_size);
+
+        // Guarantee at least some wiggle room, even if there is none from the file cache
+        let max_remaining = (total_mem as f64 * (1.0 - self.cgroup_min_overhead_fraction)) as u64;
```

Comment:
It's unclear to me what's wiggling here. Please consider improving the comment on the config variable and this ocmment.

Reply:
improved the comment here. config variable will be removed along with support for tmpfs in a follow-up PR